pkg/apiserver/patch/patch.go

```diff
@@ -0,0 +1,454 @@
+/*
```

Comment:
Clayton's point that this is generic is correct - this is a library that could (and maybe should) live outside of kube. As a result, unless we're going to discourage apiserver from using things outside of itself, I think util is the right place. If it had any dependency or coupling at all to apiserver I'd definitely say it should stay in the apiserver package.